streaming/pekko/src/main/scala/sttp/openai/streaming/pekko/package.scala

```diff
@@ -0,0 +1,51 @@
+package sttp.openai.streaming
+
+import org.apache.pekko.stream.scaladsl.{Flow, Source}
+import org.apache.pekko.util.ByteString
+import sttp.capabilities.pekko.PekkoStreams
+import sttp.client4.StreamRequest
+import sttp.model.sse.ServerSentEvent
+import sttp.openai.OpenAI
+import sttp.openai.OpenAIExceptions.OpenAIException
+import sttp.openai.json.SttpUpickleApiExtension.deserializeJsonSnake
+import sttp.openai.requests.completions.chat.ChatChunkRequestResponseData.ChatChunkResponse
+import sttp.openai.requests.completions.chat.ChatRequestBody.ChatBody
+import sttp.client4.pekkohttp.PekkoHttpServerSentEvents
+
+package object pekko {
+  import ChatChunkResponse.DoneEventMessage
+
+  implicit class extension(val client: OpenAI) {
+
+    /** Creates and streams a model response as chunk objects for the given chat conversation defined in chatBody.
+      *
+      * [[https://platform.openai.com/docs/api-reference/chat/create]]
+      *
+      * @param chatBody
+      *   Chat request body.
+      */
+    def createStreamedChatCompletion(
+        chatBody: ChatBody
+    ): StreamRequest[Either[OpenAIException, Source[ChatChunkResponse, Any]], PekkoStreams] =
+      client
+        .createChatCompletion(PekkoStreams, chatBody)
+        .mapResponse(mapEventToResponse)
+  }
+
+  private def mapEventToResponse(
+      response: Either[OpenAIException, Source[ByteString, Any]]
+  ): Either[OpenAIException, Source[ChatChunkResponse, Any]] =
+    response.map(
+      _.via(PekkoHttpServerSentEvents.parse)
+        .via(deserializeEvent)
+    )
+
+  private def deserializeEvent: Flow[ServerSentEvent, ChatChunkResponse, Any] =
+    Flow[ServerSentEvent].collect {
+      case ServerSentEvent(Some(data), _, _, _) if data != DoneEventMessage =>
```

Comment:
** issue ** This makes me thinking - the `DoneEventMessage` is simply filtered here, and we basically rely on the stream end to be signalised. Maybe we should rather `takeWhile` instead of `collect`, which would guarantee that `DoneEventMessage` completes the stream right away.
For that, we may need a test where there are more messages after `DoneEventMessage`, it's unexpected but we would be sure that "done" means "done" for us, no matter what happens.

Reply:
Similarly with `fs2` and `zio`, these modules may also need such an improvement.

Reply:
Thanks for that suggestion, I added takeWhile to the Pekko implementation